src/components/Datatable/ControlModule/ControlModule.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { getColor } from '../../../utils/helpers';
+import { FlexContainer } from '../../FlexContainer';
+import { SearchBar } from '../../forms/SearchBar';
+import { ToolsTabItem } from '../ToolsTabItem';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
+import { ControlModuleProps } from './ControlModule.types';
+import { Filters } from '../../Filters';
+import { renderSuggestionFilter } from '../../forms/SearchBar/SearchSuggestionFormats';
+
+const FiltersContainer = styled(FlexContainer)`
+  background-color: ${getColor('graphite5H')};
+`;
+
+const ControlModule: React.FC<ControlModuleProps> = ({
+  hasSearch = true,
+  searchConfig,
+  hasFiltering = true,
+  filtersConfig,
+  defaultIsFilteringOpen = true,
+  hasColumnVisibility = true,
+  // defaultHiddenColumns = [],
+  hasColumnOrdering = true,
+  // defaultColumnOrder,
+  hasCustomViews = true,
+  // defaultCustomView,
+  columnVisibilityActions,
+  filteringActions,
+  columnOrderingActions,
+  customViewsActions,
+}) => {
+  const [areFiltersOpen, setAreFiltersOpen] = useState(defaultIsFilteringOpen);
+
+  const handleCloseFilter = () => {
+    filtersConfig.onClose();
+    setAreFiltersOpen(false);
+  };
+
+  const filterToolActions = {
+    onToolActivate: () => {
+      setAreFiltersOpen(true);
+      filteringActions.onToolActivate();
+    },
+    onToolDeactivate: () => {
+      setAreFiltersOpen(false);
+      filteringActions.onToolDeactivate();
+    },
+  };
+
+  return (
+    <FlexContainer flexDirection="column">
+      <FlexContainer
+        alignItems="center"
+        padding={{ vertical: 0.4, horizontal: 0.8 }}
+      >
+        <FlexContainer flexShrink={1} margin={{ right: 1.6 }}>
+          {hasColumnVisibility && (
+            <ToolsTabItem
+              iconName={SSCIconNames.reorder}
+              label="Show/Hide"
+              {...columnVisibilityActions}
+            />
+          )}
+          {hasFiltering && (
+            <ToolsTabItem
+              iconName={SSCIconNames.filter}
+              label="Filter"
+              toolState={defaultIsFilteringOpen ? 'active' : 'default'}
+              {...filterToolActions}
+            />
+          )}
+          {hasColumnOrdering && (
```

Comment:
Grouping and Column ordering are two different features. Grouping API wasn't yet even proposed.
Column ordering is part of the `Show/Hide` feature.

Reply:
Thanks! was confused by the split, now added `hasGrouping`